src/main.ts

```diff
@@ -30,9 +30,16 @@ Vue.use(Vue2Filters) // needed by SbcFeeSummary
 
 // main code
 async function start () {
-  // fetch config from environment and API
-  // must come first as inits below depend on config
-  await fetchConfig()
+  // get config from environment
+  const processEnvBaseUrl: string = process.env.BASE_URL // eg, /business/
+  const windowLocationPathname = window.location.pathname // eg, /business/CP1234567/...
+  const windowLocationOrigin = window.location.origin // eg, http://localhost:8080
+  const applicationUrl = windowLocationOrigin + processEnvBaseUrl
+  await setBaseRouteAndBusinessId(windowLocationPathname, processEnvBaseUrl, windowLocationOrigin) // will throw error
+
+  // fetch the store first as it has no dependencies
+  const store = getVuexStore()
+  await store.dispatch('fetchConfiguration', applicationUrl)
```

Comment:
Do you have update insight into whether dispatch or commit need to be awaited? (Thinking back to that unit test that failed when `wrapper.vm.$store.commit('someMutation', someVal)` wasn't awaited.

Reply:
My understanding is that actions are asynchronous by default so the caller needs to either await the result or use `.then()` before invoking any logic that needs the result.   Mutations (aka: commits) must be synchronous.  See:  https://v3.vuex.vuejs.org/guide/mutations.html#mutations-must-be-synchronous

Reply:
Ah, so the commit is async but the mutation it calls is sync?

Or maybe there's something _weird_ going on with that unit test 🤷‍♂️ I don't usually like to ignore things like this that I don't understand 🤓

Reply:
Commits call the mutation so they *should* both be sync.  Can you send me a link to your test again?  I wonder if there's something else going on.

Reply:
https://github.com/bcgov/business-create-ui/blob/531f63ef1c5f1662a1b46f095f426ae1edcf8631/tests/unit/NameTranslation.spec.ts#L50

Reply:
If I remove the `await` and immediately call a related getter, line 8 passes.  I think this demonstrates that the mutation take place instantaneously.  

![Screenshot (27)](https://user-images.githubusercontent.com/25233684/218861743-d1a6ea0a-2175-4f3a-a567-88a3a3724ada.png)

Reply:
Maybe it needs a `await Vue.nextTick()` for some validation or reactivity to update...?

Reply:
But, weird, async shouldn't do anything on methods that aren't async, so _theoretically_ awaiting line 50 should have no effect. Except it does.

Don't spend too long on this, though I am curious why it's behaving like this.

Reply:
Yes, confirmed, adding `await Vue.nextTick()` allows the entire test to pass.  Looking quickly at "NameRequestInfo.vue", `hasNameTranslation` doesn't instantly update when the store is updated.

Reply:
Great, thanks for looking into it.

Reply:
I'll follow your lead in awaiting `store.dispatch()` and not `store.commit()` calls.